File: 7. Reverse Integer/solution.cpp
#include <iostream>
#include <string>
#include <algorithm>
#include <climits>

class Solution {
public:
    int reverse(int x) {
        // Determine the sign of the number
        int numSign = (x < 0) ? -1 : 1;
        
        // Convert the absolute value of the number to a string
        std::string str = std::to_string(std::abs(x));
        
        // Reverse the string
        std::reverse(str.begin(), str.end());
        
        // Convert the string back to an integer
        long long newNum = std::stoll(str);
        
        // Check for overflow
        if (newNum > INT_MAX) {
            return 0;
        }
        
        // Return the reversed number with the original sign
        return static_cast<int>(newNum) * numSign;
    }
};

File: 7. Reverse Integer/solution.js
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function(x) {
    var numSign = Math.sign(x);
    var str = Math.abs(x).toString();
    var newStr = str.split("").reverse().join("");

    var newNum = Number(newStr);

    if(newNum > 2**31 - 1){
        return 0;
    }

    return newNum * numSign;
};